### PR TITLE
Responsive UI, Common Key validation and hash check, Download Activity Log.

### PR DESCRIPTION
**Pull request by Cereal-Killa**

### CODE REVIEW

**Comment by FoxMcloud5655** [comment]
Wow!  You're really putting some good work into this program.  Let me know when you are finished implimenting code, and I'll look it over.

I think it's definitely time for a new update.

File: NUSGrabberGUI/Form1.cs [hunk 1]
```diff
@@ -193,49 +199,60 @@ private void STTitleList_SelectedIndexChanged(object sender, EventArgs e)
             {
                 if (v != null && v != "v0" && v != "" && v != " ")
                 {
-                    try { STVersionList.Items.Add(v.TrimStart(' ').Substring(0, v.LastIndexOf('(') - 1)); }
```
Comment (Cereal-Killa) [inline NUSGrabberGUI/Form1.cs:194]: The exception thrown here was causing the system titles to take a long time to filter.

File: NUSGrabberGUI/Form1.cs [hunk 2]
```diff
@@ -191,49 +199,60 @@ private void STTitleList_SelectedIndexChanged(object sender, EventArgs e)
...
                 }
             }
             STVersionList.SelectedIndex = 0;
             try { STTitleIDLabel.Text = "Title ID: " + (STTitleList.SelectedItem as ListItem).Title_ID.ToString(); }
             catch { }
+            STVersionList.EndUpdate();
         }
 
         private void STSearchBox_TextChanged(object sender, EventArgs e)
         {
```
Comment (Cereal-Killa) [inline NUSGrabberGUI/Form1.cs]: It looks like there were a lot of changes here however it's just the try catch that made the git diff to look huge, all I did was to add a try/catch block to block the UI and the beginupdate()/endupdate() calls.

**Review by Cereal-Killa — COMMENTED** [review]
Some comments.

**Comment by Cereal-Killa** [comment]
I screwed up reverting a commit which contained other changes, trying to recall the reverted commit now.

**Comment by Cereal-Killa** [comment]
I think I got it back, it looks cleaner now.

**Comment by Cereal-Killa** [comment]
Added a few more things...

1. Optimized the initial list parsing (was incredibly slow in development due to the many thrown exceptions).

2. If the folder is not empty and can't be cleared we'll open it using Explorer to make the user's life easier.

3. Added a download log window, basic initial version.

**Comment by Cereal-Killa** [comment]
Btw, I'm done for this pull, thanks for taking some time to review.

**Comment by FoxMcloud5655** [comment]
There's a lot of work to be done to this code to make it ready for deployment (code structure, optimizing, language versatility, debug log writing, etc.), but these are great!

I will start working on merging this and have it ready for deployment soon.

**Comment by FoxMcloud5655** [comment]
I ended up not really liking all the changes...  Sorry about that.  I'm still happy you thought it was good enough to edit on your own, though :)

**Pull request closed without merging**